src/middlewared/middlewared/plugins/failover_/disabled_reasons.py

```diff
@@ -165,6 +172,13 @@ def get_remote_reasons(self, app, ifaces, reasons):
             mismatch_nics = self.middleware.call_sync('failover.mismatch_nics')
             if mismatch_nics['missing_local'] or mismatch_nics['missing_remote']:
                 reasons.add(DisabledReasonsEnum.MISMATCH_NICS.name)
+
+            with contextlib.suppress(Exception):
+                # We do this safely because other node might not even have this method implemented
+                # if on different versions
+                if self.middleware.call_sync('failover.call_remote', 'failover.disabled.get_systemdataset_state'):
```

Comment:
This can block for 60 seconds or more. Please review mismatch_disks method or mismatch_nics method for how to work around this. No need for contextlib either.